src/Visualization/Python/PlotSlice.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import logging
+import os
+from typing import Sequence
+
+import click
+import matplotlib.pyplot as plt
+import numpy as np
+import rich
+
+import spectre.IO.H5 as spectre_h5
+from spectre.IO.Exporter import interpolate_to_points
+from spectre.Visualization.OpenVolfiles import (
+    open_volfiles_command,
+    parse_point,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def points_on_slice(
+    slice_origin: Sequence[float],
+    slice_radius: float,
+    slice_normal: Sequence[float],
+    slice_up: Sequence[float],
+    num_points: int,
+):
+    """Returns points on a rectangular slice in 3D.
+
+    Parameters:
+      slice_origin: Coordinates of the center of the slice.
+      slice_radius: Visible radius around the center of the slice.
+      slice_normal: Direction of the normal of the slice.
+      slice_up: Up-direction of the slice.
+      num_points: Number of points to return in each dimension.
+
+    Returns: An array of shape (3, num_points, num_points) with uniformly
+      spaced points on the slice.
+    """
+    u = slice_up / np.linalg.norm(slice_up)
+    n = slice_normal / np.linalg.norm(slice_normal)
+    assert np.isclose(np.dot(u, n), 0.0), "Up and normal must be orthogonal"
+    v = np.cross(u, n)
+    lower_left = slice_origin - slice_radius * (u + v)
+    affine_parameter = np.linspace(0, 1, num_points)
+    xx, yy = np.meshgrid(affine_parameter, affine_parameter)
+    return (
+        lower_left[:, np.newaxis, np.newaxis]
+        + 2.0
+        * slice_radius
+        * v[:, np.newaxis, np.newaxis]
+        * xx[np.newaxis, :, :]
+        + 2.0
+        * slice_radius
+        * u[:, np.newaxis, np.newaxis]
+        * yy[np.newaxis, :, :]
+    )
+
+
+@click.command(name="slice")
+@open_volfiles_command
+@click.option(
+    "--output",
+    "-o",
+    type=click.Path(file_okay=True, dir_okay=False, writable=True),
+    help=(
+        "Name of the output plot file. If unspecified, the plot is "
+        "shown interactively, which only works on machines with a "
+        "window server."
+    ),
+)
+@click.option(
+    "--slice-origin",
+    "-C",
+    callback=parse_point,
+    help=(
+        "Coordinates of the center of the slice through the volume "
+        "data. Specify as comma-separated list, e.g. '0,0,0'."
+    ),
+)
+@click.option(
+    "--slice-radius",
```

Comment:
Calling this `slice-radius` when it's a rectangle is a bit confusing. Maybe `slice-extents`? Or `slice-range`? Something like that.

Reply:
Ok I tried something else, see fixup.